File: Leetcode/sbgo/Leetcode318-最大单词长度乘积.go
package sbgo

func maxProduct(words []string) int {
	n := len(words)
	bitmask := make([]int, n)
	for i := 0; i < n; i++ {
		for _, ch := range words[i] {
			t := ch - 'a'
			bitmask[i] |= 1 << t
		}
	}
	res := 0
	for i := 0; i < n; i++ {
		for j := i + 1; j < n; j++ {
			if bitmask[i]&bitmask[j] == 0 {
				res = max(res, len(words[i])*len(words[j]))
			}
		}
	}
	return res
}

File: Leetcode/JS/Leetcode318-最大单词长度乘积.js
/**
 * @param {string[]} words
 * @return {number}
 */
var maxProduct = function(words) {
    let n = words.length;
    let bitmask = new Array(n).fill(0);
    for(let i = 0;i < n;i++){
        for(let ch of words[i]){
            let t = ch.charCodeAt() - 'a'.charCodeAt();
            bitmask[i] |= 1 << t;
        }
    }
    let res = 0;
    for(let i = 0;i < n;i++){
        for(let j = i + 1;j < n;j++){
            if((bitmask[i] & bitmask[j]) == 0) {
                res = Math.max(res,words[i].length * words[j].length);
            }
        }
    }
    return res;
};